lib/APPRemoteControl/App.cpp

```diff
@@ -135,16 +114,40 @@ void App::loop()
         m_controlInterval.restart();
     }
 
-    m_systemStateMachine.process();
+    if ((true == m_reportTimer.isTimeout()) && (true == m_smpServer.isSynced()))
```

Comment:
can you extract the block depenended on m_smpServer.isSynced() into a sub-function ? This function is getting big.